server/app.js

```diff
@@ -37,4 +39,149 @@ io.on('connection', (socket) => {
+        else{
+            // send the current table state
+            socket.emit('tableState', JSON.stringify(tables[0].getTableState()));
+        }
+    });
+    
+    socket.on('joinTable', function(msg){ // a user wishes to join a table
+        let profile = profiles.find(profile => profile.userID === msg.userID); // get their profile
+        let tableToJoin = tables.find(table => table.tableID === msg.tableID); // get the table they wish to join
+        if(tableToJoin === undefined || tableToJoin.isTableFull() || tableToJoin.isPlayerAtTable(profile) || profile.chips < tableToJoin.buyIn){
+            return; // the user cannot join this table
+        }
+        else{
+            // the user can join the table
+            let seatID = tableToJoin.addPlayerToTable(profile, socket.id); // the table will fill a seat based on the profile and send back the seat id
+            socket.emit('yourSeatID', seatID); // send him back his tableSeatID
+            io.emit('tableState', JSON.stringify(tableToJoin.getTableState())); // send to everyone the updated table state
+            if(tableToJoin.canAGameBegin() && !tableToJoin.tableActive){ // check if we can begin a game
+                tableToJoin.tableActive = true; // the table is now active
+                beginTheGame(io, tableToJoin);
+            }
+        }
+    });
+
+    socket.on('turnDecision', function(msg){
+        if(tables[0].isItTheirTurn(msg.userID, msg.seatID, socket.id)){
+            // there is a limited number of actions a player can take based on their current state
+            // ask if they can make that move, if not then do nothing, later also disable buttons based on lack of options
+            let response = tables[0].playerAction(msg.action, msg.raiseToValue); // will assess the action and return a response
+            if(response){
+                io.emit('tableState', JSON.stringify(tables[0].getTableState()));
+            }
+            else{
+                console.log('bad move');
+            }
+            if(tables[0].showdown){// after each move we need to check if the table is ready for a showdown
+                console.log("show down begin");
+               beginShowingTheRemainingCommunityCards(io, tables[0]);
+            }
+        }
+    });
+
+});
+
+
+// a list of sample profiles already logged into the system
+let profiles = [
+    {userID: 1000, name: 'Mark123', chips: 20000, icon: 'player_icon_1'},
+    {userID: 2000, name: 'John456', chips: 40000, icon: 'player_icon_1'},
+    {userID: 3000, name: 'Luke854', chips: 60000, icon: 'player_icon_1'},
+    {userID: 4000, name: 'Jane027', chips: 80000, icon: 'player_icon_1'},
+    {userID: 5000, name: 'Stan693', chips: 10000, icon: 'player_icon_1'},
+    {userID: 6000, name: 'Mack273', chips: 20, icon: 'player_icon_1'},
+];
+// the entire list of tables
+let tables = [];
+tables.push(new PokerTable()); // initially we will have one table
+
+function beginTheGame(io, table){
```

Comment:
Can you add function comments to these functions similar to those found in the DataAccessLayer.js